---
title: "Processing and visualizing data in the tidyverse"
author: "Fabrício de Almeida Silva  @almeidasilvaf"
institute: "PGBV/UENF"
date: "May, 2021"
output:
  xaringan::moon_reader:
    lib_dir: libs
    css: ["default", "assets/venanciolab-fonts.css", "assets/venanciolab.css"]
    self_contained: false
    seal: true
    nature:
      ratio: "16:9"
      highlightStyle: github
      highlightLines: true
      countIncrementalSlides: false
      beforeInit: "https://platform.twitter.com/widgets.js"
      navigation:
        scroll: false
---

```{r share_again, echo=FALSE}
xaringanExtra::use_share_again()
```

```{r metathis, echo=FALSE}
library(metathis)
meta() %>%
  meta_name("github-repo" = "almeidasilvaf/Project_management") %>% 
  meta_social(
    title = "Project management in bioinformatics",
    description = paste(
      "Best practices for handling bioinformatics projects",
      "February 2021"
    ),
    url = "https://almeidasilvaf.github.io/Project_management/project_management.html",
    image = "https://almeidasilvaf.github.io/Project_management/cover.png",
    image_alt = paste(
      "Title slide of Project management in bioinformatics,", 
      "presented in February 2021 by Fabrício Almeida-Silva"
    ),
    og_type = "website",
    og_author = "Fabrício Almeida-Silva",
    twitter_card_type = "summary_large_image",
    twitter_creator = "@almeidasilvaf",
    twitter_site = "@almeidasilvaf"
  )
```

```{r setup, include=FALSE}
knitr::opts_chunk$set(
  echo = FALSE, warning = FALSE, message = FALSE,
  fig.width = 7, fig.height = 6, fig.align='center'
)
options(htmltools.dir.version = TRUE)
```

```{r extra}
xaringanExtra::use_panelset()
xaringanExtra::use_scribble()
xaringanExtra::use_tile_view()
```

background-image: url("https://www.tidyverse.org/images/hex-tidyverse.png")
background-position: 95% 50%
background-size: 35%

## What is the tidyverse?

<br />

.pull-left-2[
A **collection of packages** designed for data science.

All packages in the tidyverse share data structures and grammar.

To install the whole tidyverse:

```{r install_tidyverse, echo=TRUE, eval=FALSE}
install.packages("tidyverse")
```

Then, load it.

```{r load_tidyverse, echo=TRUE}
library(tidyverse)
```

]


---

background-image: url("https://rviews.rstudio.com/post/2017-06-09-What-is-the-tidyverse_files/tidyverse1.png")
background-position: 50% 80%
background-size: 90%

## Tidyverse overview

---

background-image: url("https://www.openscapes.org/img/blog/tidydata/tidydata_1.jpg")
background-size: cover

---

background-image: url("https://www.openscapes.org/img/blog/tidydata/tidydata_2.jpg")
background-size: cover

---

background-image: url("https://allisonhorst.github.io/palmerpenguins/reference/figures/lter_penguins.png")
background-position: 50% 70%
background-size: 80%

## Meet the palmer penguins!

---

background-image: url("https://allisonhorst.github.io/palmerpenguins/reference/figures/culmen_depth.png")
background-position: 50% 70%
background-size: 80%

---

## Tibbles

.panelset[
.panel[.panel-name[Definition]
Updated and stylish version of a data frame. Main differences between tibbles and data frames:

- Input type remains unchanged

- Variable names remain unchanged

- There are no row.names() for a tibble

- Tibbles print first ten rows and columns that fit on one screen
]

.panel[.panel-name[Examples]
```{r tibble_example, echo=TRUE}
library(palmerpenguins)
penguins
```
]

.panel[.panel-name[Creating a tibble]
.pull-left[
```{r create_tibble, echo=TRUE, eval=FALSE}
tibble(
  Nome = c("Eduardo", "Juliana", "Roberta"),
  Orientador = c("Fábio", "Clicia", "Vanildo"),
  Idade = c(25, 24, 24)
)
```
]

.pull-right[
```{r show_create_tibble, ref.label="create_tibble"}
```
]
]

.panel[.panel-name[Convert to tibble]
```{r convert_tibble, echo=TRUE}
as_tibble(trees)
```
]

.panel[.panel-name[Inspecting]
```{r echo=TRUE}
slice_sample(penguins, n=5)
```

Other options: `slice_head()` and `slice_tail()`.
]

.panel[.panel-name[Subsetting]
```{r subsetting, echo=TRUE}
head(penguins$species)
penguins[1, ]
```
]
]

---

class: inverse, center, middle

# Importing and exporting data

---

## Importing and exporting data

.panelset[
.panel[.panel-name[Spreadsheets]
```{r read_excel, echo=TRUE}
library(readxl)
read_excel(readxl_example("deaths.xlsx"),
  range = "arts!A5:F8",
  .name_repair = "unique")
```

Important arguments: 
- sheet: name of the sheet to import
- col_names: names for columns. By default, it is the first row.
]

.panel[.panel-name[CSVs]
**CSV** stands for **comma-separated values**.

```{r read_csv, echo=TRUE}
pokemon <- read_csv("https://gist.githubusercontent.com/armgilles/194bcff35001e7eb53a2a8b441e8b2c6/raw/92200bc0a673d5ce2110aaad4544ed6c4010f687/pokemon.csv")
head(pokemon, n=5)
```
]

.panel[.panel-name[TSVs]
**TSV** stands for **tab-separated values**.

```{r read_tsv, echo=TRUE}
xmas <- read_tsv("https://raw.githubusercontent.com/rfordatascience/tidytuesday/master/data/2019/2019-12-24/christmas_lyrics.tsv")
head(xmas, n=5)
```
]

.panel[.panel-name[Delimited files]
```{r read_delim, echo=TRUE}
xmas <- read_delim("https://raw.githubusercontent.com/rfordatascience/tidytuesday/master/data/2019/2019-12-24/christmas_lyrics.tsv",
                 delim="\t")
head(xmas, n=5)
```
]

.panel[.panel-name[Exporting]

To export data, use the `write_` functions. Example:

```{r export, echo=TRUE, eval=FALSE}
write_csv(xmas, file="christmas_songs.csv") # CSV
write_tsv(xmas, file="christmas_songs.tsv") # TSV
write_delim(xmas, file="christmas_songs.txt", delim=" ") # Space-separated
```
]
]

---

class: inverse, center, middle

# Data processing

---

## Reshaping data

.panelset[
.panel[.panel-name[Wide and long data]
```{r wide_long}
knitr::include_graphics("https://thinkr.fr/wp-content/uploads/long_wide.png")
```
]

.panel[.panel-name[Raw data]
```{r raw_data, echo=TRUE}
airquality <- as_tibble(airquality)
airquality
```
]

.panel[.panel-name[pivot_longer() and pivot_wider()]
```{r pivot_longer, echo=TRUE}
gathered <- airquality %>%
  pivot_longer(everything())

head(gathered)
```

To revert to wide format, use `pivot_wider()`.
]
]

---

## Wrangling data

.panelset[

.panel[.panel-name[filter()]

.pull-left-1[
Filter by 1 condition:

```{r filter1, echo=TRUE, eval=FALSE}
penguins %>%
  filter(sex == "female") %>%
  head(n=3)
```

Filter by 2+ conditions:

```{r filter2, echo=TRUE, eval=FALSE}
penguins %>%
  filter(sex == "female", body_mass_g > 3500) %>%
  head(n=3)
```
]

.pull-right-2[
```{r show_filter1, ref.label="filter1"}
```

```{r show_filter2, ref.label="filter1"}
```

]


]

.panel[.panel-name[select()]
```{r select, echo=TRUE}
penguins %>%
  filter(species == "Adelie") %>%
  select(bill_length_mm, bill_depth_mm)
```
]

.panel[.panel-name[arrange()]
.pull-left-1[

Sort in ascending order.

```{r arrange1, echo=TRUE, eval=FALSE}
penguins %>%
  select(species, bill_length_mm, year) %>%
  arrange(bill_length_mm) %>%
  head(n=3)
```

Sort in descending, then ascending order.

```{r arrange2, echo=TRUE, eval=FALSE}
penguins %>%
  select(species, bill_length_mm, bill_depth_mm, year) %>%
  arrange(-bill_length_mm, bill_depth_mm) %>%
  head(n=3)
```
]

.pull-right-2[
```{r show_arrange1, ref.label="arrange1"}
```

```{r show_arrange2, ref.label="arrange2"}
```
]
]

.panel[.panel-name[mutate()]

```{r mutate, echo=TRUE}
penguins %>%
  select(species, island, bill_length_mm, bill_depth_mm) %>%
  mutate(length_times_depth = bill_length_mm * bill_depth_mm)
```
]

.panel[.panel-name[separate()]

```{r load_conservation}
conservation <- read_csv("https://raw.githubusercontent.com/suzanbaert/Dplyr_Tutorials/master/conservation_explanation.csv")
```

.pull-left-2[
```{r separate, echo=TRUE, eval=FALSE}
# Show original data
head(conservation, n=3)

# Separate columns
sep <- conservation %>%
  separate(`conservation abbreviation`, 
           into = c("abbreviation", "description"), sep = " = ") %>%
  head(n=3)
sep
```
]

.pull-right-1[
```{r show_separate, ref.label="separate"}
```
]

]

.panel[.panel-name[unite()]
```{r unite, echo=TRUE}
sep %>%
  unite(united_col, abbreviation, description, sep = " = ") %>%
  head(n=3)
```
]

.panel[.panel-name[clean_names()]
```{r clean_names, echo=TRUE}
library(janitor)
conservation %>%
  clean_names()
```
]

]

---

background-image: url("https://media.giphy.com/media/ZVik7pBtu9dNS/giphy.gif")
background-position: 5% 50%
background-size: 35%

## Exercise #1

.pull-right-2[
Filter the `penguins` data set to keep only female penguins from species "Adelie"
and sort them by bill length in ascending order.

- What are the first 3 values for `bill_length_mm`?

- From which island are the top 3 Adelie penguins with the smallest bill?
]

---

## Solution - Exercise #1

```{r ex1_solution, echo=TRUE}
penguins %>%
  filter(sex == "female", species == "Adelie") %>%
  arrange(bill_length_mm)
```

---

background-image: url("https://media.giphy.com/media/ZVik7pBtu9dNS/giphy.gif")
background-position: 5% 50%
background-size: 35%

## Exercise #2

.pull-right-2[
Filter the `penguins` data set to remove observations from the year 2007. Then, select only the columns `species`, `sex`, and `body_mass_g`. Finally, create a new column named `body_mass_kg` with the body mass represented in kilograms and sort this column in descending order.

- What are weights (in kg) of the 5 heaviest penguins?

- What are their species?

- What are their sexes?

]

---

## Solution - Exercise #2

```{r solution_ex2, echo=TRUE}
penguins %>%
  filter(year != 2007) %>%
  select(species, sex, body_mass_g) %>%
  mutate(body_mass_kg = body_mass_g / 1000) %>%
  arrange(-body_mass_kg)
```

---

background-image: url("https://media.giphy.com/media/ZVik7pBtu9dNS/giphy.gif")
background-position: 5% 50%
background-size: 35%

## Exercise #3

.pull-right-2[
<br />

- How many penguins have flipper length between 200 and 215 mm?

**Tip:** you can use the `count()` or the `nrow()` function to count.
]

---

## Solution - Exercise #3

```{r solution_ex3, echo=TRUE}
penguins %>%
  filter(flipper_length_mm > 200, flipper_length_mm < 215) %>%
  count()
```

---

background-image: url("https://data-lessons.github.io/gapminder-R/fig/dplyr-joins.png")
background-size: 60%

## Joins

---

background-image: url("https://github.com/gadenbuie/tidyexplain/raw/master/images/inner-join.gif")
background-size: 60%

## Joins

---

background-image: url("httpshttps://github.com/gadenbuie/tidyexplain/raw/master/images/left-join.gif")
background-size: 60%

## Joins

---

background-image: url("https://github.com/gadenbuie/tidyexplain/raw/master/images/left-join-extra.gif")
background-size: 60%

## Joins

---

background-image: url("https://github.com/gadenbuie/tidyexplain/raw/master/images/right-join.gif")
background-size: 60%

## Joins

---

background-image: url("https://github.com/gadenbuie/tidyexplain/raw/master/images/full-join.gif")
background-size: 60%

## Joins

---

## Joining data

.panelset[

.panel[.panel-name[The data]
```{r data_joins}
superheroes <- tibble::tribble(
       ~name, ~alignment,  ~gender,          ~publisher,
   "Magneto",      "bad",   "male",            "Marvel",
     "Storm",     "good", "female",            "Marvel",
  "Mystique",      "bad", "female",            "Marvel",
    "Batman",     "good",   "male",                "DC",
     "Joker",      "bad",   "male",                "DC",
  "Catwoman",      "bad", "female",                "DC",
   "Hellboy",     "good",   "male", "Dark Horse Comics"
  )

publishers <- tibble::tribble(
  ~publisher, ~yr_founded,
        "DC",       1934L,
    "Marvel",       1939L,
     "Image",       1992L
  )
```

.pull-left[
```{r show_data_joins1, echo=TRUE}
superheroes
```
]

.pull-right[
```{r show_data_join2, echo=TRUE}
publishers
```
]
]

.panel[.panel-name[inner_join()]
```{r inner_join, echo=TRUE}
inner_join(superheroes, publishers)
```
]

.panel[.panel-name[semi_join()]
```{r semi_join, echo=TRUE}
semi_join(superheroes, publishers)
```
]

.panel[.panel-name[left_join()]
```{r left_join, echo=TRUE}
left_join(superheroes, publishers)
```
]

.panel[.panel-name[anti_join()]
```{r anti_join, echo=TRUE}
anti_join(superheroes, publishers)
```
]

.panel[.panel-name[full_join()]
```{r full_join, echo=TRUE}
full_join(superheroes, publishers)
```
]
]

---

## Summarizing data

.panelset[

.panel[.panel-name[summarize()]
.pull-left-1[
```{r summarize1, echo=TRUE, eval=FALSE}
head(msleep)
msleep %>%
  select(genus) %>%
  summarize(N=n())
```
]

.pull-right-2[
```{r show_summarize1, ref.label="summarize1"}
```
]
]

.panel[.panel-name[group_by() [1]]
.pull-left[
```{r group_by1, eval=FALSE, echo=TRUE}
msleep %>%
  group_by(order) %>% 
  select(order) %>%
  summarize(N=n())
```
]

.pull-right[
```{r show_groupby1, ref.label='group_by1'}
```
]
]

.panel[.panel-name[group_by() [2]]
.pull-left[
```{r group_by2, echo=TRUE, eval=FALSE}
msleep %>%
  group_by(order) %>% 
  select(order, sleep_total) %>%
  summarize(N=n(), mean_sleep=mean(sleep_total))
```
]

.pull-right[
```{r show_group_by2, ref.label='group_by2'}
```
]
]

.panel[.panel-name[count()]
.pull-left[
```{r count, echo=TRUE, eval=FALSE}
msleep %>%
  count(vore, order)
```
]

.pull-right[
```{r show_count, ref.label='count'}
```
]
]

.panel[.panel-name[tabyl()]
.pull-left[
```{r tabyl, echo=TRUE, eval=FALSE}
msleep %>%
  tabyl(order)
```
]

.pull-right[
```{r show_tabyl, ref.label='tabyl'}
```
]
]

]

---

---

background-image: url("https://media.giphy.com/media/ZVik7pBtu9dNS/giphy.gif")
background-position: 5% 50%
background-size: 35%

## Exercise #4

.pull-right-2[
<br />

- For each species, what is the mean body mass **(in kg)** by sex?

]

---

## Solution - Exercise #4

```{r solution_ex4, echo=TRUE}
penguins %>%
  group_by(species, sex) %>%
  summarise(mean_weight = mean(body_mass_g)) %>%
  filter(!is.na(sex)) %>%
  mutate(mean_weight = mean_weight / 1000)
```

---

## Operations across() columns

.panelset[
.panel[.panel-name[Example 1]

Use across() to apply a function to all (or user-defined) columns.

.pull-left[
```{r across, echo=TRUE, eval=FALSE}
head(airquality, n=3)
airquality %>%
  summarize(across(Ozone:Temp, mean, na.rm = TRUE))
```
]
.pull-right[
```{r show_across, ref.label="across"}
```
]
]

.panel[.panel-name[Example 2]

.pull-left[
```{r across2, echo=TRUE, eval=FALSE}
airquality %>%
  filter(across(Ozone:Temp, ~ !is.na(.)))
```
]
.pull-right[
```{r show_across2, ref.label="across2"}
```
]
]

.panel[.panel-name[Example 3]

.pull-left[
```{r across3, echo=TRUE, eval=FALSE}
airquality %>%
  mutate(across(Ozone:Temp, ~ replace_na(., 0)))
```
]
.pull-right[
```{r show_across3, ref.label="across3"}
```
]
]
]

---

## String manipulation with {stringr}

.panelset[

.panel[.panel-name[str_c()] .pull-left[
```{r str_c, echo=TRUE, eval=FALSE}
msleep %>%
  mutate(name_genus = str_c(name, genus, "_")) %>%
  select(name_genus)
```
]
.pull-right[
```{r str_cview, ref.label='str_c'}
```
]
]

.panel[.panel-name[str_sub()]
```{r str_sub, echo=TRUE}
genes <- c("Glyma.15G026400.1", "Glyma.15G026400.2", "Glyma.15G026400.3")
str_sub(genes, 1, 15)
```
]

.panel[.panel-name[str_detect()]
```{r echo=TRUE}
genes <- c("AT2G25880", "AT2G45490", "Achn130491", "ATR0705G186", "ATR0705G186")
genes[str_detect(genes, "^AT[0-9]+")]
```
]

.panel[.panel-name[str_replace()]
```{r str_replace, echo=TRUE}
genes <- c("Glyma.15G026400.1", "Glyma.15G026400.2", "Glyma.15G026400.3")
str_replace(genes, "\\.[0-9]$", "")
```
]
]

---

background-image: url("https://media.giphy.com/media/fCU27lxFIsDaRhKCe2/giphy.gif")
background-position: 5% 50%
background-size: 35%

## Exercise #5

.pull-right-2[
<br />

Load the Netflix titles data set and find the shows/movies starring Millie Bobby Brown.

- How many are they?

- What are their names?

Link [here](https://raw.githubusercontent.com/rfordatascience/tidytuesday/master/data/2021/2021-04-20/netflix_titles.csv).

]

---

## Solution - Exercise #5

```{r ex5, echo=TRUE}
netflix <- readr::read_csv("https://raw.githubusercontent.com/rfordatascience/tidytuesday/master/data/2021/2021-04-20/netflix_titles.csv")
netflix %>%
  filter(str_detect(cast, "Millie Bobby Brown")) %>%
  select(title)
```

---

background-image: url("https://media.giphy.com/media/ZVik7pBtu9dNS/giphy.gif")
background-position: 5% 50%
background-size: 35%

## Exercise #6

.pull-right-2[
<br />

From the netflix title data set, count the number of titles per country.

- What are the top 3 countries with the largest number of movies/shows?

- Where is Brazil in this ranking?

**Tip:** you will need the separate_rows() function.
]


---

## Solution - Exercise #6

.pull-left[
```{r ex6, echo=TRUE, eval=FALSE}
ranking <- netflix %>%
  separate_rows(country, sep = ", ") %>%
  count(country) %>%
  arrange(-n)

head(ranking, n=3)

ranking %>%
  mutate(Position = 1:nrow(ranking)) %>%
  filter(country == "Brazil")
```
]
.pull-right[
```{r show_ex6, ref.label='ex6'}
```
]

---

class: inverse, middle, center

# Data visualization with ggplot2

---

background-image: url("https://qph.fs.quoracdn.net/main-qimg-834d36ac56c16413f8948c4b9b3ca0df")
background-size: 95%

---

## On plots

.panelset[
.panel[.panel-name[Good plots]
Good plots have: 

- Clearly-labeled axes.

- Text that are large enough to see.

- Axes that are not misleading.

- Data that are displayed appropriately considering the type of data you have.
]

There are **exploratory plots** and **explanatory plots**.

.panel[.panel-name[Exploratory plots]
These plots help us **identify patterns and understand them**. Some of their features:

- they are made quickly

- you should not be concerned about axes labels, titles, etc.

- they are made all the time during analyses
]

.panel[.panel-name[Explanatory plots]
These plots are used for **communicating information with others**. Things to consider:

- They may take a lot of time to be produced

- Typically, you will make just a few of them

]
]

---

background-image: url("https://mk0rfortheresto0o08q.kinstacdn.com/wp-content/uploads/2019/07/grammar-of-graphics.png")
background-size: 95%

## The grammar of graphics

.footnote[Source: David - R for the Rest of Us]

---

background-image: url("https://image.slidesharecdn.com/grammarofgraphicsdvny-181102042310/95/grammar-of-graphics-the-secret-sauce-of-powerful-data-stories-15-638.jpg?cb=1541133222")
background-size: 95%

## The grammar of graphics

.footnote[Source: Ganes Kesari]

---

background-image: url("figs/aes.png")
background-size: 95%

## Common aesthetics

.footnote[Source: Claus Wilke]

---

## ggplot2 syntax

.pull-left[
```{r ggplot_1, echo=TRUE, eval=FALSE}
ggplot(penguins, aes(x = bill_length_mm, 
                     y = body_mass_g))
```
]
.pull-right[
```{r ggplot_1_show, ref.label='ggplot_1'}
```
]

---

## ggplot2 syntax

.pull-left[
```{r ggplot_2, echo=TRUE, eval=FALSE}
ggplot(penguins) +
  geom_point(aes(x = bill_length_mm, 
                 y = body_mass_g))
```
]
.pull-right[
```{r ggplot_2_show, ref.label='ggplot_2'}
```
]

---

## ggplot2 syntax

.pull-left[
```{r ggplot_3, echo=TRUE, eval=FALSE}
ggplot(penguins, aes(x = bill_length_mm, 
                     y = body_mass_g)) +
  geom_point()
```
]
.pull-right[
```{r ggplot_3_show, ref.label='ggplot_3'}
```
]

---

## ggplot2 syntax

.pull-left[
```{r ggplot_4, echo=TRUE, eval=FALSE}
ggplot(penguins, aes(x = bill_length_mm, y = body_mass_g)) +
  geom_point(aes(color = species))
```
]
.pull-right[
```{r ggplot_4_show, ref.label='ggplot_4'}
```
]

---

## ggplot2 syntax

.pull-left[
```{r ggplot_5, echo=TRUE, eval=FALSE}
ggplot(penguins, aes(x = bill_length_mm, y = body_mass_g)) +
  geom_point(aes(color = species)) +
  facet_wrap(~ year)
```
]
.pull-right[
```{r ggplot_5_show, ref.label='ggplot_5'}
```
]

---

## ggplot2 syntax

.pull-left[
```{r ggplot_6, echo=TRUE, eval=FALSE}
ggplot(penguins, aes(x = bill_length_mm, y = body_mass_g)) +
  geom_point(aes(color = species)) +
  facet_wrap(~ year) +
  theme_classic()
```
]
.pull-right[
```{r ggplot_6_show, ref.label='ggplot_6'}
```
]

---

## ggplot2 syntax

.pull-left[
```{r ggplot_7, echo=TRUE, eval=FALSE}
ggplot(penguins, aes(x = bill_length_mm, y = body_mass_g)) +
  geom_point(aes(color = species)) +
  facet_wrap(~ year) +
  theme_bw()
```
]
.pull-right[
```{r ggplot_7_show, ref.label='ggplot_7'}
```
]

---

## ggplot2 syntax

.pull-left[
```{r ggplot_8, echo=TRUE, eval=FALSE}
ggplot(penguins, aes(x = bill_length_mm, y = body_mass_g)) +
  geom_point(aes(color = species)) +
  facet_wrap(~ year) +
  theme_bw() +
  scale_colour_brewer(palette="Set2")
```
]
.pull-right[
```{r ggplot_8_show, ref.label='ggplot_8'}
```
]

---

## ggplot2 syntax

.pull-left[
```{r ggplot_9, echo=TRUE, eval=FALSE}
ggplot(penguins, aes(x = bill_length_mm, y = body_mass_g)) +
  geom_point(aes(color = species)) +
  facet_wrap(~ year) +
  theme_bw() +
  scale_colour_brewer(palette="Set2") +
  labs(x="Bill length (mm)",
       y="Body mass(g)",
       color="Species")
```
]
.pull-right[
```{r ggplot_9_show, ref.label='ggplot_9'}
```
]

---

## ggplot2 syntax

.pull-left[
```{r ggplot_10, echo=TRUE, eval=FALSE}
ggplot(penguins, aes(x = bill_length_mm, y = body_mass_g)) +
  geom_point(aes(color = species)) +
  facet_wrap(~ year) +
  theme_bw() +
  scale_colour_brewer(palette="Set2") +
  labs(x="Bill length (mm)",
       y="Body mass(g)",
       color="Species") +
  ggtitle("Bill length vs body mass per year")
```
]
.pull-right[
```{r ggplot_10_show, ref.label='ggplot_10'}
```
]

---

## ggplot2 syntax

.pull-left[
```{r ggplot_11, echo=TRUE, eval=FALSE}
ggplot(penguins, aes(x = bill_length_mm, y = body_mass_g)) +
  geom_point(aes(color = species)) +
  facet_wrap(~ year) +
  theme_bw() +
  scale_colour_brewer(palette="Set2") +
  labs(x="Bill length (mm)",
       y="Body mass(g)",
       color="Species") +
  ggtitle("Bill length vs body mass per year") +
  theme(
    plot.title = element_text(size=13, hjust=0.5)
  )
```
]
.pull-right[
```{r ggplot_11_show, ref.label='ggplot_11'}
```
]

---

## Scatterplots

**Data:** Two continuous variables. They are used to explore relationships.

.pull-left[
```{r scatterplot, echo=TRUE, eval=FALSE}
ggplot(penguins, aes(x = bill_length_mm, y = body_mass_g)) +
  geom_point(aes(color = species)) +
  facet_wrap(~ year) +
  theme_bw() +
  scale_colour_brewer(palette="Set2") +
  labs(x="Bill length (mm)",
       y="Body mass(g)",
       color="Species") +
  ggtitle("Bill length vs body mass per year") +
  theme(
    plot.title = element_text(size=13, hjust=0.5)
  )
```
]
.pull-right[
```{r scatterplot_show, ref.label='scatterplot'}
```
]

---

## Barplot

**Data:** counts of a single categorical variable.

.pull-left[
```{r barplot1, echo=TRUE, eval=FALSE}
ggplot(penguins) +
  geom_bar(aes(year))
```
]
.pull-right[
```{r show_barplot1, ref.label='barplot1'}
```
]

---

## Barplot

**Data:** counts of a single categorical variable.

.pull-left[
```{r barplot2, echo=TRUE, eval=FALSE}
ggplot(penguins) +
  geom_bar(aes(year, fill=year)) +
  ylim(0, 150) +
  labs(x="Year", y="Count") +
  ggtitle("Frequency of penguins per year") +
  theme_bw()
```
]
.pull-right[
```{r show_barplot2, ref.label='barplot2'}
```
]

---



---

## Histograms (geom_histogram)

**Data:** a single continuous variable.

.pull-left[
```{r histogram, echo=TRUE, eval=FALSE}
ggplot(penguins, aes(body_mass_g)) +
  geom_histogram()
```
]

.pull-right[
```{r show_histogram, ref.label='histogram'}
```
]

---

## Histograms (geom_histogram)

**Data:** a single continuous variable.

.pull-left[
```{r histogram2, echo=TRUE, eval=FALSE}
penguins %>%
  filter(!is.na(sex)) %>%
  ggplot(., aes(body_mass_g)) +
  geom_histogram(bins = 40, color='black') +
  theme_bw() +
  ggtitle("Distribution of body mass (in g)") +
  labs(x="Body mass (g)", y="Count")
```
]

.pull-right[
```{r show_histogram2, ref`.label='histogram2'}
```
]

---

## Density plots (geom_density)

**Data:** a single continuous variable. They are a smooth version of histograms, and they are not dependent on the number of bins.

.pull-left[
```{r density1, echo=TRUE, eval=FALSE}
penguins %>%
  filter(!is.na(sex)) %>%
  ggplot(., aes(body_mass_g)) +
  geom_density(fill="grey") +
  theme_bw() +
  ggtitle("Distribution of body mass (in g)") +
  labs(x="Body mass (g)", y="Count")
```
]

.pull-right[
```{r show_density1, ref.label='density1'}
```
]

---

## Box plots

---

class: sydney-yellow, middle, center

## Here's where you can find me:

`r icons::fontawesome("twitter")`  [@almeidasilvaf](https://twitter.com/almeidasilvaf)

`r icons::fontawesome("github")`  [almeidasilvaf](https://github.com/almeidasilvaf/)

`r icons::fontawesome("globe")`  [almeidasilvaf.github.io/home](https://almeidasilvaf.github.io/home/)

`r icons::fontawesome("envelope")`  [fabricio_almeidasilva@hotmail.com](mailto:fabricio_almeidasilva@hotmail.com)

`r icons::academicons("orcid")`  [Fabricio Almeida-Silva](https://orcid.org/0000-0002-5314-2964)

